File: Medium/Longest Palindrome in a String/longest-palindrome-in-a-string.py
#User function Template for python3

class Solution:
    def longestPalin(self, S):
        res=""
        s2=0
        if(S==S[::-1]):
            return (S)
        for i in range(len(S)):
            st1=''
            s1=0
            for j in range(i,len(S)):
                st1+=S[j]
                s1+=1
                if(st1==st1[::-1]):
                    if(s1>s2):
                        res=st1
                        s2=s1
        return res
        

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':

    t = int(input())

    for _ in range(t):
        S = input()

        ob = Solution()

        ans = ob.longestPalin(S)

        print(ans)
# } Driver Code Ends

File: Difficulty: Medium/Longest Palindrome in a String/longest-palindrome-in-a-string.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    
    bool isPalindrome(string &s){
        int i = 0;
        int j = s.size()-1;
        while(i < j){
            if(s[i] != s[j]){
                return false;
            }
            i++;
            j--;
        }
        
        return true;
    }
  
    string longestPalindrome(string &s) {
        int n = s.size();
        string ans = "";
        for(int i=0;i<n;i++){
            for(int j=n-1;j>=i;j--){
                if(s[i] == s[j]){
                    string temp = s.substr(i,j-i+1);
                    if(isPalindrome(temp)){
                        if(ans.size() < temp.size()){
                            ans = temp;
                        }
                        break;
                    }
                }
            }
        }
        
        return ans;
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        string S;
        cin >> S;

        Solution ob;
        cout << ob.longestPalindrome(S) << endl;

        cout << "~"
             << "\n";
    }
}

// } Driver Code Ends